charts/surrealdb/templates/deployment.yaml

```diff
@@ -56,8 +56,20 @@ spec:
             - name: SURREAL_PASS
               value: {{ . }}
             {{- end }}
+            {{- $auth := "uninitialized" }}
+            {{- if hasKey .Values.surrealdb "auth" }}
+            {{- if hasKey .Values.surrealdb "unauthenticated" }}
+            {{ fail "surrealdb.auth and surrealdb.unauthenticated are mutually exclusive" }}
```

Comment:
This implies you must use `helm upgrade --reset-values --set surrealdb.unauthenticated=true` if you're moving from the legacy default value `surrealdb.auth=false` to the new `surrealdb.unauthenticated=true`.
Is that ok for everyone..?

I consider always passing `--reset-values` is the best practice anyway.